pkg/koordlet/util/meminfo.go

```diff
@@ -114,13 +142,95 @@ func readMemInfo(path string) (*MemInfo, error) {
 	return &info, nil
 }
 
+func GetMemInfo() (*MemInfo, error) {
+	memInfoPath := system.GetProcFilePath(system.ProcMemInfoName)
+	memInfo, err := readMemInfo(memInfoPath, false)
+	if err != nil {
+		return nil, err
+	}
+	return memInfo, nil
+}
+
 // GetMemInfoUsageKB returns the node's memory usage quantity (kB)
 func GetMemInfoUsageKB() (int64, error) {
 	meminfoPath := system.GetProcFilePath(system.ProcMemInfoName)
-	memInfo, err := readMemInfo(meminfoPath)
+	memInfo, err := readMemInfo(meminfoPath, false)
 	if err != nil {
 		return 0, err
 	}
 	usage := int64(memInfo.MemTotal - memInfo.MemAvailable)
 	return usage, nil
 }
+
+type NUMAInfo struct {
+	NodeID  int32    `json:"nodeID,omitempty"`
+	MemInfo *MemInfo `json:"memInfo,omitempty"`
+}
+
+// NodeNUMAInfo represents the node NUMA information.
+// Currently, it just contains the meminfo for each NUMA node.
+type NodeNUMAInfo struct {
+	NUMAInfos  []NUMAInfo         `json:"numaInfos,omitempty"`
+	MemInfoMap map[int32]*MemInfo `json:"memInfoMap,omitempty"` // nodeID -> memInfo
+}
+
+// GetNodeNUMAInfo gets the node NUMA information with the pre-configured sysfs path.
+func GetNodeNUMAInfo() (*NodeNUMAInfo, error) {
+	numaNodeParentDir := system.GetSysNUMADir()
+	nodeDirs, err := os.ReadDir(numaNodeParentDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read NUMA dir, err: %w", err)
+	}
+
+	result := &NodeNUMAInfo{
+		MemInfoMap: map[int32]*MemInfo{},
+	}
+	maxNodeID := int32(-1)
+	for _, n := range nodeDirs {
+		dirName := n.Name() // assert string pattern `nodeX`
+		if len(dirName) < 4 || dirName[:4] != "node" {
+			klog.V(5).Infof("failed to get node NUMA info, err: invalid dir name %s", dirName)
```

Comment:
these error should be v4 if the prefix of dir is node

Reply:
done